### ISSUE TITLE
Linux: ShowFrameNumber crashes on YV12 clip

### ISSUE THREAD

**Issue by magiblot** [AviSynth/AviSynthPlus#247]
When I play the following script on Linux:

```py
BlankClip(pixel_type="YV12")
ShowFrameNumber
```

I get the following crash:

```
/usr/include/c++/11.1.0/bits/stl_vector.h:1045: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](std::vector<_Tp, _Alloc>::size_type) [with _Tp = short unsigned int; _Alloc = std::allocator<short unsigned int>; std::vector<_Tp, _Alloc>::reference = short unsigned int&; std::vector<_Tp, _Alloc>::size_type = long unsigned int]: Assertion '__n < this->size()' failed.

Thread 8 "mpv/opener" received signal SIGABRT, Aborted.
[Switching to Thread 0x7fffe1165640 (LWP 13656)]
0x00007ffff50a5d22 in raise () from /usr/lib/libc.so.6
(gdb) bt
#0  0x00007ffff50a5d22 in raise () at /usr/lib/libc.so.6
#1  0x00007ffff508f862 in abort () at /usr/lib/libc.so.6
#2  0x00007fffcb5d727a in std::__replacement_assert(char const*, int, char const*, char const*)
    (__file=__file@entry=0x7fffcb8a2608 "/usr/include/c++/11.1.0/bits/stl_vector.h", __line=__line@entry=1045, __function=__function@entry=0x7fffcb9a2af0 "std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](std::vector<_Tp, _Alloc>::size_type) [with _Tp = short unsigned int; _Alloc = std::allocator<short unsigned int>; std::vector<_"..., __condition=__condition@entry=0x7fffcb8a27cd "__n < this->size()")
    at /usr/include/c++/11.1.0/x86_64-pc-linux-gnu/bits/c++config.h:504
#3  0x00007fffcb640132 in std::vector<unsigned short, std::allocator<unsigned short> >::operator[](unsigned long)
    (this=<optimized out>, this=<synthetic pointer>, __n=<optimized out>) at /usr/include/c++/11.1.0/bits/stl_vector.h:1045
#4  do_DrawStringPlanar<8, false, false>(int, int, unsigned char**, int*, int, int, int, BitmapFont const*, int, int, std::vector<int, std::allocator<int> >&, int, int, int, bool)
    (width=width@entry=640, height=height@entry=480, dstps=dstps@entry=0x7fffe1163b10, pitches=pitches@entry=0x7fffe1163ad4, logXRatioUV=logXRatioUV@entry=1, logYRatioUV=logYRatioUV@entry=1, planeCount=<optimized out>, bmfont=<optimized out>, x=<optimized out>, y=<optimized out>, s=<optimized out>, color=<optimized out>, halocolor=<optimized out>, align=<optimized out>, useHalocolor=<optimized out>) at /usr/src/debug/avisynthplus/avs_core/core/info.cpp:928
#5  0x00007fffcb62c932 in DrawString_internal(BitmapFont*, VideoInfo const&, PVideoFrame&, int, int, std::wstring&, int, int, bool, int, bool)
    (current_font=0x7fffd02463f0, vi=..., dst=<optimized out>, x=639, y=24, s16=<optimized out>, color=13766802, halocolor=1081472, useHalocolor=true, align=3, fadeBackground=false) at /usr/src/debug/avisynthplus/avs_core/core/info.cpp:1410
#6  0x00007fffcb80a270 in ShowFrameNumber::GetFrame(int, IScriptEnvironment*) (this=0x7fffd005a490, n=0, env=<optimized out>)
    at /usr/src/debug/avisynthplus/avs_core/filters/text-overlay.cpp:1163
#7  0x00007fffcb5eb2fb in MTGuard::GetFrame(int, IScriptEnvironment*) (this=0x7fffd0245790, n=0, env=0x7fffd005a390)
    at /usr/src/debug/avisynthplus/avs_core/core/MTGuard.cpp:125
#8  0x00007fffcb624554 in Cache::GetFrame(int, IScriptEnvironment*) (this=0x7fffd0018d80, n=0, env_=0x7fffd005a390)
    at /usr/src/debug/avisynthplus/avs_core/core/cache.cpp:216
#9  0x00007fffcb6242f7 in CacheGuard::GetFrame(int, IScriptEnvironment*) (this=0x7fffd003c300, n=0, env=<optimized out>)
    at /usr/src/debug/avisynthplus/avs_core/core/cache.cpp:505
#10 0x00007fffcb621980 in avs_get_frame(AVS_Clip*, int) (p=0x7fffd0018db0, n=<optimized out>)
    at /usr/src/debug/avisynthplus/avs_core/core/avisynth_c.cpp:832
#11 0x00007ffff63934ef in  () at /usr/lib/libavformat.so.58
#12 0x00007ffff6393885 in  () at /usr/lib/libavformat.so.58
#13 0x00007ffff64e1e1a in  () at /usr/lib/libavformat.so.58
#14 0x00007ffff64e287b in  () at /usr/lib/libavformat.so.58
#15 0x00007ffff64ea67b in avformat_find_stream_info () at /usr/lib/libavformat.so.58
#16 0x00005555555c3e99 in  ()
#17 0x00005555555bb7da in  ()
#18 0x00005555555bc0b3 in  ()
#19 0x000055555561dc41 in  ()
#20 0x00007ffff7ced259 in start_thread () at /usr/lib/libpthread.so.0
#21 0x00007ffff51675e3 in clone () at /usr/lib/libc.so.6
```

Other pixel types do not seem to be affected (`YUV422P`, `YUV444P`, `YUY2`...).

Using current master (https://github.com/AviSynth/AviSynthPlus/commit/0b08a30e3f540e3413711f8b107bb2e02ffd1031).

**Comment by pinterf** [comment]
Thanks, reproduced, fix is in progress

**Comment by magiblot** [comment]
Thank you! No hurry.

**Comment by magiblot** [comment]
Fixed by 1ed2116a895a94c34670d31f95406be6d3808722.
